net/dns/resolver/forwarder.go

```diff
@@ -598,6 +598,17 @@ func (f *forwarder) forwardWithDestChan(ctx context.Context, query packet, respo
 	// out, playing on Sonos still works.
 	if hasRDNSBonjourPrefix(domain) {
 		metricDNSFwdDropBonjour.Add(1)
+		res, err := nxDomainResponse(query)
+		if err != nil {
+			f.logf("error parsing bonjour query: %v", err)
+			return nil
+		}
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case responseChan <- res:
```

Comment:
Should we block here, or just drop the response?

Reply:
I think it's fine... this is exactly what the packet writing code at the bottom does, and the caller's timeout is 5 seconds. So worst case it blocks for 5 seconds if userspace engine is having a bad day reading these out.